packages/core/src/providers/lastfm.ts

```diff
@@ -0,0 +1,189 @@
+/**
+ * <div class="provider" style={{backgroundColor: "#000", display: "flex", justifyContent: "space-between", color: "#fff", padding: 16}}>
+ * <span>Built-in <b>Last.fm</b> integration.</span>
+ * <a href="https://www.last.fm/">
+ *   <img style={{display: "block"}} src="https://authjs.dev/img/providers/lastfm.svg" height="48" />
+ * </a>
+ * </div>
+ *
+ * @module providers/lastfm
+ */
+import { createHash } from "crypto";
+
+import type { TokenSet } from "../types.js";
+import type { OAuthConfig, OAuthUserConfig } from "./index.js";
+
+export interface LastfmSignatureParams {
+  method: string;
+  token: string;
+  api_key: string;
+  api_secret: string;
+}
+
+export interface LastfmSession {
+  session: {
+    name: string;
+    key: string;
+    subscriber: string;
+  };
+}
+
+export interface LastfmProfileImage {
+  size: "small" | "medium" | "large" | "extralarge";
+  "#text": string;
+}
+
+export interface LastfmProfile extends Record<string, any> {
+  user: {
+    name: string;
+    age: string;
+    subscriber: string;
+    realname: string;
+    bootstrap: string;
+    playcount: string;
+    artist_count: string;
+    playlists: string;
+    track_count: string;
+    album_count: string;
+    image: LastfmProfileImage[];
+    registered: {
+      unixtime: string;
+      "#text": number;
+    };
+    country: string;
+    gender: string;
+    url: string;
+    type: string;
+  };
+}
+
+function generateApiSig({
+  method,
+  token,
+  api_key,
+  api_secret,
+}: LastfmSignatureParams): string {
+  const raw = `api_key${api_key}method${method}token${token}${api_secret}`;
+
+  return createHash("md5").update(raw).digest("hex");
+}
+
+/**
+ * Add Last.fm login to your page.
+ *
+ * ### Setup
+ *
+ * #### Callback URL
+ * ```
+ * https://example.com/api/auth/callback/lastfm
+ * ```
+ *
+ * #### Configuration
+ *```ts
+ * import { Auth } from "@auth/core"
```

Comment:
Quick note here - as a user you should never have to interact with or install `@auth/core`. 

So these imports should both come from the client lib the user is using, i.e. `next-auth` for next.js or `@auth/sveltekit` for sveltekit, etc.